python-package/xgboost/spark/core.py

```diff
@@ -315,6 +320,23 @@ def _validate_params(self):
                     )
 
 
+def _validate_and_convert_feature_col_as_float_col(
+    dataset, features_col_name: list
+) -> list:
+    """feature column names must be IntegralType or float or double types"""
+    feature_cols = []
+    for c in features_col_name:
+        if isinstance(dataset.schema[c].dataType, DoubleType):
+            feature_cols.append(col(c).cast(FloatType()).alias(c))
+        elif isinstance(dataset.schema[c].dataType, (FloatType, IntegralType)):
```

Comment:
What's IntegralType type ? Do you mean IntegerType ?

Reply:
@trivialfis, could you correct me if xgboost supports IntegralType (Byte/Integer/Long/Short) ?

Reply:
Yes, sort of. XGBoost will convert the data to float internally.

Reply:
BTW, XGBoost can convert it on the fly without creating a copy of data.

Reply:
Thx @trivialfis, Yeah, if the type is short/byte, it will reduce the size of the shuffle write.